src/equations/compressible_navier_stokes_1d.jl

```diff
@@ -159,10 +163,16 @@ function flux(u, gradients, orientation::Integer,
     # in the implementation
     q1 = equations.kappa * dTdx
 
-    # Constant dynamic viscosity is copied to a variable for readability.
-    # Offers flexibility for dynamic viscosity via Sutherland's law where it depends
-    # on temperature and reference values, Ts and Tref such that mu(T)
-    mu = equations.mu
+    # In the simplest cases, the user passed in `mu` or `mu()` 
+    # (which returns just a constant) but
+    # more complex functions like Sutherland's law are possible.
+    if equations.mu isa Real
+        mu = equations.mu
+    else
+        # The equations are equipped with a function that computes the dynamic viscosity mu
+        # from the current state. 
+        mu = equations.mu(u, equations)
+    end
```

Comment:
IMHO, this is somewhat inconvenient, since it essentially requires the user to add this `if` statement anywhere that `mu` is used. Also, should the compiler not be able to optimize away the `if` statement, you have a potential branch here, which is not optimal for performance.

What about adding something like this (names tbd):
```julia
dynamic_viscosity(u, equations) = dynamic_viscosity(u, equations.mu, equations)
dynamic_viscosity(u, mu::Real, equations) = mu
dynamic_viscosity(u, mu::T, equations) where {T} = mu(u, equations)
```
That would allow the user to just use `dynamic_viscosity(u, equations)` in their code (including here in `flux` and it would do the right thing depending on whether `mu` is a real value or something else?

Reply:
Nice, yeah this is far more elegant :+1:

Reply:
Nice!